core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java

```diff
@@ -47,6 +47,24 @@ public interface InstanceOperations {
   void setProperty(final String property, final String value)
       throws AccumuloException, AccumuloSecurityException;
 
+  /**
+   * Sets multiple system properties in zookeeper. Tablet servers will pull this setting and
+   * override the equivalent setting in accumulo.properties. Changes can be seen using
+   * {@link #getSystemConfiguration()}.
+   * <p>
+   * Only some properties can be changed by this method, an IllegalArgumentException will be thrown
+   * if there is an attempt to set a read-only property.
+   *
+   * @param propertiesMap
+   *          map containing key-value pairs of properties and corresponding values
+   * @throws AccumuloException
+   *           if a general error occurs
+   * @throws AccumuloSecurityException
+   *           if the user does not have permission
+   */
+  void setProperties(Map<String,String> propertiesMap)
```

Comment:
From the description of this new method, it's not clear if the set of properties will be merged with the existing properties that are set in ZooKeeper, or if they will completely replace them. If they are to be merged, then this API should also include the ability to mark a property to be deleted at the same time that other properties are set. So, I think it's probably easier to just make this map replace any existing map that is currently stored in ZooKeeper.

(In other words, if there was an existing override set, and it's not duplicated or overridden in this map here, then that previous override will be removed)

Reply:
It may be easier to adopt the addOrUpdate semantics - there may be default properties that are added on table creation - and you probably do not want to remove them unless it is done explicitly.

I'm not sure how addOrUpdate handles empty (or null) values - but that could be used to remove a property as part of an atomic set group operation.  Doing something like thatMap.of("a", "v1", "b", ", "c", "v2")  - would either add or update a or c and remove b.

Reply:
The three primitives are add, update, and remove. However, "add or update" is just "put" (or "set"), and "remove" isn't really a removal, but a "revert to default" (removal of any override set at this scope). I definitely want to include the ability to remove a property alongside the properties to set/update. I don't know if it's intuitive to rely on setting it to a null value, since that *looks like* it's setting an override value of `null`. I figured it was more intuitive to omit it from the map, and replace the entire map with the provided one, since having a full on Mutation API with `putDeletes` and such is a bit overkill for config.

Reply:
Maybe the API should accept a Map of properties to set, and a Set of properties to revert to default? That might make more intuitive than relying on interpreting null values. If a property name is in both the map and the set, the client can throw an IllegalArgumentException. On the server side, the current map could be mutated atomically by removing anything from the set, and then putting anything from the map, and storing the result.